lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -0,0 +1,200 @@
+      self.display_name = 'Reach'
+      self.currencies_without_fractions = %w(BIF BYR CLF CLP CVE DJF GNF ISK JPY KMF KRW PYG RWF UGX UYI VND VUV XAF XOF XPF IDR MGA MRO)
+
+      API_VERSION = 'v2.21'.freeze
+      STANDARD_ERROR_CODE_MAPPING = {}
+      PAYMENT_METHOD_MAP = {
+        american_express: 'AMEX',
+        cabal: 'CABAL',
+        check: 'ACH',
+        dankort: 'DANKORT',
+        diners_club: 'DINERS',
+        discover: 'DISC',
+        elo: 'ELO',
+        jcb: 'JCB',
+        maestro: 'MAESTRO',
+        master: 'MC',
+        naranja: 'NARANJA',
+        visa: 'VISA'
+      }
+
+      def initialize(options = {})
+        requires!(options, :merchant_id, :secret)
+        super
+      end
+
+      def authorize(money, payment, options = {})
+        request = build_checkout_request(money, payment, options)
+        add_custom_fields_data(request, options)
+        add_customer_data(request, options, payment)
+
+        post = { request: request, card: add_payment(payment) }
+        commit('checkout', post)
+      end
+
+      def purchase(money, payment, options = {})
+        options[:capture] = true
+        authorize(money, payment, options)
+      end
+
+      def capture(money, authorization, options = {})
+        merchant_id, order_id = authorization.split '#'
+        post = { request: { MerchantId: merchant_id, OrderId: order_id } }
+
+        commit('capture', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+      end
+
+      private
+
+      def build_checkout_request(amount, payment, options)
+        {
+          MerchantId: @options[:merchant_id],
+          ReferenceId: options[:order_id],
+          ConsumerCurrency: options[:currency] || currency(options[:amount]),
+          Capture: options[:capture] || false,
+          PaymentMethod: PAYMENT_METHOD_MAP[payment.brand.to_sym],
+          Items: [
+            Sku: SecureRandom.alphanumeric,
+            ConsumerPrice: amount,
+            Quantity: 1
+          ],
+          ViaAgent: true # Indicates this is server to server API call
+        }
+      end
+
+      def add_payment(payment)
+        {
+          Name: payment.name,
+          Number: payment.number,
+          Expiry: { Month: payment.month, Year: payment.year },
+          VerificationCode: payment.verification_value
+        }
+      end
+
+      def add_customer_data(request, options, payment)
+        address = options[:billing_address] || options[:address]
+
+        request[:Consumer] = {
+          Name: payment.respond_to?(:name) ? payment.name : address[:name],
+          Email: options[:email],
+          Address: address[:address1],
+          City: address[:city],
+          Country: address[:country]
+        }.compact
+      end
+
+      def add_custom_fields_data(request, options)
+        if options[:fingerprint].present?
+          request[:DeviceFingerprint] = options[:fingerprint]
```

Comment:
🤔 
Hi @jherreraa I'm wondering if the DeviceFingerprint field should be sent through other kind of transaction as the capture, according to this part of the docs
<img width="1264" alt="Screen Shot 2022-11-04 at 3 05 56 PM" src="https://user-images.githubusercontent.com/1335028/200065102-f97597d2-6b7f-48f8-986c-7a3d01c765e9.png">

Reply:
Hello! 
According to reach API documentation this field is required in the checkout request which is the base to the other requests and capture request does not require the field.